spec/lib/registries/topic_taxonomy_registry_spec.rb

```diff
@@ -0,0 +1,94 @@
+require "securerandom"
+require 'spec_helper'
+require "gds_api/test_helpers/content_store"
+
+RSpec.describe Registries::TopicTaxonomyRegistry do
+  include ::GdsApi::TestHelpers::ContentStore
+
+  let(:content_id_one) { "top-level-taxon-one" }
+  let(:content_id_two) { "top-level-taxon-two" }
+  let(:top_level_taxon_one) { top_level_taxon(content_id_one) }
+  let(:top_level_taxon_two) { top_level_taxon(content_id_two) }
+
+  describe "when topic taxonomy API is unavailable" do
+    it "will return an (uncached) empty hash" do
+      clear_cache
+      topic_taxonomy_api_is_unavailable
+      expect(described_class.new[content_id_one]).to be_nil
+      expect(Rails.cache.fetch(described_class::CACHE_KEY)).to be_nil
+    end
+  end
+
+
+  describe "when topic taxonomy api is available" do
+    before :each do
+      clear_cache
+      topic_taxonomy_has_taxons
+    end
+
+    after { clear_cache }
+
+    subject(:registry) { described_class.new }
+
+    it "will fetch an expanded topic taxon by content_id" do
+      fetched_document = registry[content_id_one]
+      expect(fetched_document['content_id']).to eq(top_level_taxon_one['content_id'])
+      expect(fetched_document['title']).to eq(top_level_taxon_one['title'])
+
+      fetched_document = registry[content_id_two]
+      expect(fetched_document['content_id']).to eq(top_level_taxon_two['content_id'])
+      expect(fetched_document['title']).to eq(top_level_taxon_two['title'])
+    end
+  end
+
+  def topic_taxonomy_api_is_unavailable
+    stub_request(:get, topic_taxonomy_endpoint).to_return(status: 500)
+  end
+
+  def topic_taxonomy_has_taxons
+    stub_request(:get, topic_taxonomy_endpoint).
```

Comment:
Could use content_store_has_item from the gds-api-adapers test helpers:  https://github.com/alphagov/gds-api-adapters/blob/1dadd526f523a744413e166f42c4cd3f01e3084b/lib/gds_api/test_helpers/content_store.rb#L22